src/Security/Jwt/Validator/Validator.php

```diff
@@ -32,7 +32,7 @@ class Validator implements ValidatorInterface
     /** @var ConfigurationFactory */
     private $factory;
 
-    public function __construct(ConfigurationFactory $factory = null)
+    public function __construct(?ConfigurationFactory $factory = null)
```

Comment:
Btw, can we make an interface for this factory? Otherwise, when we need to update ruleset it requires to replace both factory and validator(

Reply:
- the ruleset wont change, it's guided by IMS specs
- the abstraction layer is actually done on parser, validator and builder (interfaces), those default implementations just work with lcobucci in a way that dont need change